upload_vehicle_data.py

```diff
@@ -0,0 +1,380 @@
+"""This script automates the upload of rosbags from vehicle to cloud host."""
+
+import argparse
+import logging
+import os
+import subprocess
+
+import yaml
+
+# Setup logging
+logging.basicConfig(
+    filename='upload_vehicle_data.log',
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)s: %(message)s',
+)
+
+
+def run_ssh_command(remote_user, remote_ip, command):
+    """Run a command on the remote machine using SSH."""
+    ssh_command = f"ssh {remote_user}@{remote_ip} '{command}'"
+    try:
+        subprocess.run(ssh_command, shell=True, check=True)
+        logging.info(f'Ran SSH command: {command}')
+    except subprocess.CalledProcessError as e:
+        logging.error(f'Failed to run SSH command: {command}: {e}')
+        raise
+
+
+def create_remote_directory(remote_user, remote_ip, remote_directory):
+    """Create a directory on the remote machine."""
+    command = f'mkdir -p {remote_directory}/temp'
+    run_ssh_command(remote_user, remote_ip, command)
+
+
+def delete_remote_directory(remote_user, remote_ip, remote_directory):
+    """Delete a directory on the remote machine."""
+    command = f'rm -rf {remote_directory}/temp'
+    run_ssh_command(remote_user, remote_ip, command)
+
+
+def delete_remote_directory_contents(remote_user, remote_ip, remote_directory):
+    """Delete the contents of a directory on the remote machine."""
+    command = f'rm -rf {remote_directory}/*'
+    run_ssh_command(remote_user, remote_ip, command)
+
+
+def compress_and_transfer_rosbag(
+    remote_user,
+    remote_ip,
+    rosbag_path,
+    remote_directory,
+    cloud_upload_directory,
+):
+    """Compress rosbag on remote machine and transfer it to cloud host."""
+    remote_temp_directory = f'{remote_directory}/temp'
+    # Check available disk space before compression
+    if not check_disk_space(
+        remote_user, remote_ip, remote_temp_directory, rosbag_path
+    ):
+        logging.error(f'Insufficient disk space for compressing {rosbag_path}')
+        return False
+
+    # Compress the rosbag on the remote machine
+    remote_compressed_path = os.path.join(
+        remote_temp_directory, os.path.basename(rosbag_path)
+    )
+    mcap_path = '/home/mcap'

```

Comment:
You will need to use the `$HOME` env variable instead. This variable will give you `/home/<username>` path instead. This this script should look for llowing  path `/home/<username>/mcap`. You can read env variables from python with :
```suggestion
    mcap_path= os.path.join(os.getenv('HOME'), 'mcap')
```

Reply:
changed